src/main/java/com/ge/predix/uaa/token/lib/DefaultZoneConfiguration.java

```diff
@@ -25,18 +25,19 @@
 import org.springframework.util.AntPathMatcher;
 
 public class DefaultZoneConfiguration implements InitializingBean {
-    private String trustedIssuerId;
     private List<String> allowedUriPatterns;
     private List<String> trustedIssuerIds = new ArrayList<>();
 
+    // This field is for backward compatibility.
     public String getTrustedIssuerId() {
-        return this.trustedIssuerId;
+        return this.trustedIssuerIds.isEmpty() ? null : this.trustedIssuerIds.get(0);
     }
 
     public void setTrustedIssuerId(final String trustedIssuerId) {
-        this.trustedIssuerId = trustedIssuerId;
-        // providing default behavior for existing single issuer contract
-        this.trustedIssuerIds.add(trustedIssuerId);
+        if (null != trustedIssuerId) {
+            this.trustedIssuerIds = new ArrayList<>(); // replace list
+            this.trustedIssuerIds.add(trustedIssuerId);
+        }
```

Comment:
To be consistent with setTrustedIssuerIds, replace existing list if null is passed.

Reply:
done